diff --git src/mrb_geoip.c src/mrb_geoip.c
index dd60c29..c9ccd0a 100644
--- src/mrb_geoip.c
+++ src/mrb_geoip.c
@@ -203,6 +203,14 @@ static mrb_value mrb_geoip_time_zone(mrb_state *mrb, mrb_value self)
           data->gir->country_code, data->gir->region)));
 }
 
+static mrb_value mrb_geoip_clear(mrb_state *mrb, mrb_value self)
+{
+  mrb_geoip_data *data = DATA_PTR(self);
+  GeoIPRecord_delete(data->gir);
+  GeoIP_delete(data->gi);
+  return mrb_fixnum_value(0);
+}
+
 void mrb_mruby_geoip_gem_init(mrb_state *mrb)
 {
     struct RClass *geoip;
@@ -220,6 +228,7 @@ void mrb_mruby_geoip_gem_init(mrb_state *mrb)
     mrb_define_method(mrb, geoip, "metro_code", mrb_geoip_metro_code, MRB_ARGS_NONE());
     mrb_define_method(mrb, geoip, "area_code", mrb_geoip_area_code, MRB_ARGS_NONE());
     mrb_define_method(mrb, geoip, "time_zone", mrb_geoip_time_zone, MRB_ARGS_NONE());
+    mrb_define_method(mrb, geoip, "clear", mrb_geoip_clear, MRB_ARGS_NONE());
     DONE;
 }
 
